client/app/scripts/superdesk-publish/publish.js

```diff
@@ -167,35 +176,50 @@
             });
         }));
 
-        promises.push(adminPublishSettingsService.fetchFormattedItems().then(function(items) {
-            $scope.formattedItems = items._items;
-            _.each(items._items, function(item) {
-                $scope.formattedItemLookup[item._id] = item;
-            });
-        }));
-
         function fetchPublishQueue () {
             var criteria = criteria || {};
             criteria.max_results = 200;
             criteria.sort = '[(\'_created\',-1)]';
+
+            if ($scope.selectedFilterSubscriber !== null && $scope.selectedFilterChannel !== null) {
+                criteria.where = JSON.stringify({
+                        '$and': [
+                            {'output_channel_id': $scope.selectedFilterChannel._id},
+                            {'subscriber_id': $scope.selectedFilterSubscriber._id}
+                        ]
+                    });
+            } else if ($scope.selectedFilterChannel !== null) {
+                criteria.where = {'output_channel_id': $scope.selectedFilterChannel._id};
+            } else if ($scope.selectedFilterSubscriber !== null) {
+                criteria.where = {'subscriber_id': $scope.selectedFilterSubscriber._id};
+            }
+
             return api.publish_queue.query(criteria);
         }
 
         $scope.reload = function() {
             $q.all(promises).then(function() {
                 fetchPublishQueue().then(function(queue) {
-                    $scope.publish_queue = queue._items;
+                    var queuedItems = queue._items;
+
+                    _.forEach(queuedItems, function(item) {
+                        angular.extend(item, {'selected': false});
+                    });
+
+                    $scope.publish_queue = queuedItems;
                     $scope.lastRefreshedAt = new Date();
+
+                    $scope.showResendBtn = false;
+                    $scope.showCacnelBtn = false;
                 });
             });
         };
 
-        $scope.scheduleToSend = function(item) {
+        $scope.buildNewSchedule = function (item) {
             var newItem = {};
```

Comment:
There are other properties other than `published_seq_num` like error_message,...in future there can be some others, that's the reason am building a new object.